### PR TITLE
Fix non gfx10 build

### PR DESCRIPTION
**Pull request by dnovillo**
This adds extra guards around code that is only valid when building with
GFX10 enabled.

Tested with GFX10 support disabled and GFX10 support enabled.

Note that this requires changes in the other drivers as well.

### CODE REVIEW

File: patch/gfx9/chip/llpcGfx9Chip.cpp [hunk 1]
```diff
@@ -30,6 +30,8 @@
  */
 #define DEBUG_TYPE "llpc-gfx9-chip"
 
+#include <stdio.h>
+
```
Comment (nhaehnle) [inline patch/gfx9/chip/llpcGfx9Chip.cpp]: Why is that required?

**Comment by dnovillo** [comment]
> Mostly LGTM, though we should probably just drop all those #ifdefs wholesale instead.

Yeah, I think most of the code is dynamically protected for GFX10, so it makes better sense to me.   I don't think I have a gfx9 part to test things on, however.

File: patch/gfx9/chip/llpcGfx9Chip.cpp (hunk 1 above)
Comment (dnovillo) [reply patch/gfx9/chip/llpcGfx9Chip.cpp]: Oh, this one was for the calls to `snprintf`.  Interestingly, it only happens with our internal build system (clang-based).  I didn't really compare the pre-processed output to figure out why.

```
patch/gfx9/chip/llpcGfx9Chip.cpp:849:26: error: use of undeclared identifier 'snprintf'; did you mean 'vswprintf'?
        int32_t length = snprintf(unknownRegNameBuf, 256, "UNKNOWN(0x%08X)", regId);
                         ^~~~~~~~
                         vswprintf
/usr/include/wchar.h:628:12: note: 'vswprintf' declared here
extern int vswprintf (wchar_t *__restrict __s, size_t __n,
           ^
patch/gfx9/chip/llpcGfx9Chip.cpp:849:35: error: cannot initialize a parameter of type 'wchar_t *' with an lvalue of type 'char [256]'
        int32_t length = snprintf(unknownRegNameBuf, 256, "UNKNOWN(0x%08X)", regId);
                                  ^~~~~~~~~~~~~~~~~
/usr/include/wchar.h:628:43: note: passing argument to parameter '__s' here
extern int vswprintf (wchar_t *__restrict __s, size_t __n,
                                          ^
2 errors generated.
```

**Comment by amdrexu** [comment]
We probably need to clear GFX10 macro in the near future. It is not required and recent code changes might break non-GFX10 build because developers tend to forget to add the macro.

File: patch/gfx9/chip/llpcGfx9Chip.cpp (hunk 1 above)
Comment (nhaehnle) [reply patch/gfx9/chip/llpcGfx9Chip.cpp]: Huh, interesting. We do have one Clang-based builder internally that didn't show this. But yeah, it's obviously a reasonable change, I just got suspicious because it wasn't mentioned in the commit message.

**Comment by dnovillo** [comment]
> We probably need to clear GFX10 macro in the near future. It is not required and recent code changes might break non-GFX10 build because developers tend to forget to add the macro.

It would be very useful to have a bot that built the common variants.  @kuhar was thinking of adding some bots for the configurations we are interested in.  Jakub, could you include a build that disables GFX10 (`-DXGL_BUILD_GFX10=OFF` in the `cmake` call).

File: patch/gfx9/chip/llpcGfx9Chip.cpp (hunk 1 above)
Comment (dnovillo) [reply patch/gfx9/chip/llpcGfx9Chip.cpp]: Yeah, sorry about that.  I was in auto-pilot ;)

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by dnovillo** [comment]
Thanks!
